source/fundamentals/typescript.txt

```diff
@@ -79,81 +80,51 @@ You can find a code snippet that shows how to specify a type for the ``FindCurso
-
-     The following code snippet does not raise a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-no-error
-        :end-before: end-no-error
-
-  .. tab:: Nested Objects
-     :tabid: nested-objects
-
-     The following code snippet raises a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-error
-        :end-before: end-error
-
-     This is the error:
+   .. tab:: Recursive Type
+      :tabid: recursive-type
 
-     .. code-block:: text
+      .. code-block:: typescript
 
-        Type 'string' is not assignable to type 'number'.
+         interface RecursiveType {
+            r: RecursiveType | BaseCase;
+            i: number;
+         }
 
-Despite the lack of type safety, we still recommend that you use dot notation to
-access nested fields in query and update documents when you use TypeScript. You
-must manually check that your nested field values have your intended type.
+         type BaseCase = "done"; 
 
-.. note:: Reason To Use Dot Notation
+   .. tab:: Code Snippet
+      :tabid: code-snippet
 
-   In the MongoDB Query Language, you must match a subdocument exactly
-   when specifying subdocuments in a query. Dot notation allows you to query
-   nested fields without matching subdocuments exactly.
+      The following code snippet uses a recursive type:
 
-   To show this behavior, lets say you have a collection containing
-   only the following document:
+      .. code-block:: typescript
 
-   .. code-block:: json
+         const database = client.db("test-db");
+         const col = database.collection<RecursiveType>("test-collection");
+         const document = await col.findOne({ i: 1 });
 
-      { field: { s1: "hi", s2: "bye" } }
+      The preceding code snippet raises the following error at compile time:
 
-   The following query returns no results from this collection, as the value of
-   ``field`` does not exactly match ``{ s1: "hi" }``:
+      .. code-block:: text
 
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-no-doc
-      :end-before: end-no-doc
-
-   The following queries both return your document:
-
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-doc
-      :end-before: end-doc
+         error TS2615: Type of property 'r' circularly references itself in mapped type '{ [Key in keyof RecursiveType]: [Key, ...NestedPaths<RecursiveType[Key]>]; }'.
 
-   The syntax of the query that does not use dot notation is cumbersome and hard
-   to understand, and may not be worth the type safety obtained from
-   avoiding dot notation.
+If you must use a recursive type to specify a schema for your documents,
```

Comment:
Simplify the sentence for clarity:
"If you must use a recursive type in your document schema..."